cluster/juju/layers/kubernetes-master/reactive/kubernetes_master.py

```diff
@@ -215,32 +230,35 @@ def setup_leader_authentication():
     # eg:
     # {'/root/cdk/serviceaccount.key': 'RSA:2471731...'}
     charms.leadership.leader_set(leader_data)
-
     set_state('authentication.setup')
 
 
 @when_not('leadership.is_leader')
-@when_not('authentication.setup')
 def setup_non_leader_authentication():
```

Comment:
Could removing the `@when_not('authentication.setup')` decorator here could result in a 5 minute delay?:

1. `setup_non_leader_authentication` handler runs
2. `password_changed` handler runs, removes `authentication.setup` state
3. reactive does not dispatch `setup_non_leader_authentication` handler again until the next hook

I'm having trouble reasoning through the specific outcome of that, but we've had similar problems with delays caused by using `is_state` in combination with state-based dispatch.